mula/scheduler/app.py

```diff
@@ -252,8 +252,41 @@ def run(self) -> None:
             interval=self.ctx.config.monitor_organisations_interval,
         )
 
+        # Start metrics collecting
+        self.run_in_thread(
+            name="metrics_collector",
+            func=self.collect_metrics,
+            interval=1,
+        )
+
         # Main thread
         while not self.stop_event.is_set():
             time.sleep(0.01)
 
         self.shutdown()
+
+    def collect_metrics(self) -> None:
+        """Collect application metrics
+
+        This method that allows to collect metrics throughout the application.
+        """
+        # Collect metrics from the schedulers
+        for s in self.schedulers.values():
+            self.ctx.metrics_qsize.labels(
+                scheduler_id=s.scheduler_id,
+            ).set(
+                s.queue.qsize(),
+            )
+
+        # Collect metrics from the katalogus
+        k = self.ctx.services.katalogus
+        self.ctx.metrics_katalogus.info(
+            {
+                "plugin_cache_expiry_time": k.organisations_plugin_cache.start_time.isoformat(),
```

Comment:
These should be separate metrics, not labels on a single metric. Using labels with high cardinality values creates separate time series that can't be analysed but do result in high storage costs.

A good example of a metric and label combination would be `http_requests_total{method="$method"}` where `$method` could be `GET`, `POST` or `DELETE` and the value of the metric is the total number of http requests for each method.

https://prometheus.io/docs/practices/naming/#labels has more information about metrics and labels usage.

Reply:
Cool, thanks! I'll have a look at it.

Reply:
@praseodym I've made some changes and updated the example output at the top. Would this be the correct way to do it?

Reply:
Yes that looks a lot better! There's no metric for `scheduler_qsize` now though?

Reply:
Ok cool, thanks! Ah yeah I see. Think it is because I haven't added an organisation yet. But I'll make sure that's the case.

Reply:
Yeah it had to do with an organisation not yet being created, when it has been created the following output will be generated by the endpoint:

```
HTTP/1.1 200 OK
content-length: 1088
content-type: text/plain; charset=utf-8
date: Thu, 06 Apr 2023 07:05:42 GMT
server: uvicorn

# HELP app_settings_info Scheduler configuration settings
# TYPE app_settings_info gauge
app_settings_info{monitor_organisations_interval="60",pq_maxsize="1000",pq_populate_grace_period="86400",pq_populate_interval="60",pq_populate_max_random_objects="50"} 1.0
# HELP scheduler_qsize Size of the scheduler queue
# TYPE scheduler_qsize gauge
scheduler_qsize{scheduler_id="normalizer-dev"} 0.0
scheduler_qsize{scheduler_id="boefje-dev"} 0.0
# HELP app_katalogus_info Katalogus service information
# TYPE app_katalogus_info gauge
app_katalogus_info 1.0
# HELP katalogus_cache_expiry Katalogus cache expiry
# TYPE katalogus_cache_expiry gauge
katalogus_cache_expiry{cache_type="plugin"} 1.680764699e+09
katalogus_cache_expiry{cache_type="boefje"} 1.68076476e+09
katalogus_cache_expiry{cache_type="normalizer"} 1.68076476e+09
# HELP katalogus_cache_start Katalogus cache start
# TYPE katalogus_cache_start gauge
katalogus_cache_start{cache_type="plugin"} 1.680764669e+09
katalogus_cache_start{cache_type="boefje"} 1.680764669e+09
katalogus_cache_start{cache_type="normalizer"} 1.680764669e+09

```